fips.Dockerfile

```diff
@@ -0,0 +1,132 @@
+# check=skip=RedundantTargetPlatform
+# setup build image
+FROM --platform=$TARGETPLATFORM mcr.microsoft.com/oss/go/microsoft/golang:1.24.2-fips-bullseye@sha256:28ab4742d3b5feb0b3c7450629b4e105128d8709dd6ca22898472ab302140c37 AS operator-build
+
+ENV GOEXPERIMENT=systemcrypto
+
+WORKDIR /app
+
+ARG DEBUG_TOOLS
+RUN if [ "$DEBUG_TOOLS" = "true" ]; then \
+      GOBIN=/app/build/_output/bin go install github.com/go-delve/delve/cmd/dlv@latest; \
+    fi
+
+COPY go.mod go.sum ./
+RUN go mod download -x
+
+COPY pkg ./pkg
+COPY cmd ./cmd
+
+ARG GO_LINKER_ARGS
+ARG GO_BUILD_TAGS
+ARG TARGETARCH
+ARG TARGETOS
+
+RUN --mount=type=cache,target="/root/.cache/go-build" \
+    --mount=type=cache,target="/go/pkg" \
+    CGO_ENABLED=1 GOFIPS=1 GOOS=$TARGETOS GOARCH=$TARGETARCH \
+    go build -tags "${GO_BUILD_TAGS}" -trimpath -ldflags="${GO_LINKER_ARGS}" \
+    -o ./build/_output/bin/dynatrace-operator ./cmd/
+
+# platform is required, otherwise the copy command will copy the wrong architecture files, don't trust GitHub Actions linting warnings
+FROM --platform=$TARGETPLATFORM registry.access.redhat.com/ubi9-micro:9.5-1744118077@sha256:dca8bc186bb579f36414c6ad28f1dbeda33e5cf0bd5fc1c51430cc578e25f819 AS base
+FROM --platform=$TARGETPLATFORM registry.access.redhat.com/ubi9:9.5-1744101466@sha256:ea57285741f007e83f2ee20423c20b0cbcce0b59cc3da027c671692cc7efe4dd AS dependency
+RUN mkdir -p /tmp/rootfs-dependency
+COPY --from=base / /tmp/rootfs-dependency
+RUN dnf install --installroot /tmp/rootfs-dependency \
+      util-linux-core tar \
+      --releasever 9 \
+      --setopt install_weak_deps=false \
+      --nodocs -y \
+ && dnf --installroot /tmp/rootfs-dependency clean all \
+ && rm -rf \
+      /tmp/rootfs-dependency/var/cache/* \
+      /tmp/rootfs-dependency/var/log/dnf* \
+      /tmp/rootfs-dependency/var/log/yum.*
+
+# install openssl-dependencies
+RUN dnf install --setopt install_weak_deps=false --nodocs -y make gcc perl
+
+WORKDIR /openssl_build
+# build and install openssl
+# version must be FIPS certified, details https://openssl-library.org/source/
+# get the sha256 from trusted source (e.g. Github release, https://github.com/openssl/openssl/releases)
+ENV OPENSSL_BUILD_VERSION="3.1.2"
+ENV OPENSSL_BUILD_TARBALL_SHA256="a0ce69b8b97ea6a35b96875235aa453b966ba3cba8af2de23657d8b6767d6539"
+ENV OPENSSL_BUILD_CONFIGURE_ARGS="enable-fips"
+
+RUN curl -L -o src.tgz https://github.com/openssl/openssl/releases/download/openssl-${OPENSSL_BUILD_VERSION}/openssl-${OPENSSL_BUILD_VERSION}.tar.gz && \
+    sha256sum --quiet -c - <<< "${OPENSSL_BUILD_TARBALL_SHA256}  src.tgz" && \
+    tar --strip-components=1 -xzf src.tgz
```

Comment:
```suggestion
    tar --strip-components=1 -xzf openssl-${OPENSSL_BUILD_VERSION}.tar.gz
```

Reply:
why?
IMO it makes no sense, just makes the thing longer

Reply:
atleast rename `src` -> `openssl.tar.gz`.
It's hard to read, and it confuses what is src?)